icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -155,6 +155,7 @@ void RBBITest::runIndexedTest( int32_t index, UBool exec, const char* &name, cha
     TESTCASE_AUTO(TestBug22585);
     TESTCASE_AUTO(TestBug22602);
     TESTCASE_AUTO(TestBug22636);
+    TESTCASE_AUTO(TestBug22849);
```

Comment:
This test case shouldn't be part of this PR - it's for a different issue.

Reply:
it is the way I saw the memory leak. I do not know what better way to show that memroy leak.

Reply:
This test case passes also without the bugfix, so it doesn't verify whether the bugfix works or not.

That said, I don't think this change needs 3 different reviewers, so I'll remove myself.

Reply:
Poking into this test a little bit, it turns out that not only does the run time go exponential with the number of '.'s in the rule pattern, but the size of the generated state table also goes exponential.

Maybe the fuzzer's docker environment has a lowish total heap size limit, which caused the allocation failure, showing the bug. But the test does not fail in a normal development environment, and I question whether it would fail in CI testing.

The size of the allocation triggered by this test is 526696 bytes, which is no big deal, but I don't know how much other stuff is left on the heap by the rule builder.

I still recommend removing this test. It's not going to produce consistent, reproducible results.

Reply:
ok, I remove the tests